File: src/main/java/com/chatsummary/bot/model/ChatMessage.java
package com.chatsummary.bot.model;

import java.time.Instant;
import java.util.List;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "chat_messages")
public record ChatMessage(
        @Id String id,
        @Indexed long chatId,
        Integer telegramMessageId,
        String senderName,
        String text,
        List<ChatAttachment> attachments,
        @Indexed Instant timestamp
) {

    public ChatMessage {
        attachments = attachments == null ? List.of() : List.copyOf(attachments);
        timestamp = timestamp == null ? Instant.now() : timestamp;
    }

    public ChatMessage(long chatId, Integer telegramMessageId, String senderName, String text) {
        this(null, chatId, telegramMessageId, senderName, text, List.of(), Instant.now());
    }

    public ChatMessage(long chatId, Integer telegramMessageId, String senderName, String text, List<ChatAttachment> attachments) {
        this(null, chatId, telegramMessageId, senderName, text, attachments, Instant.now());
    }
}

File: src/main/kotlin/com/chatsummary/bot/model/ChatMessage.kt
package com.chatsummary.bot.model

import org.bson.types.Binary
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.index.Indexed
import java.time.Instant

@Document(collection = "chat_messages")
data class ChatMessage(
    @Id
    val id: String? = null,

    @Indexed
    val chatId: Long,

    val senderName: String,

    val text: String,

    // Добавляем список вложенных изображений
    val attachments: List<ChatAttachment> = emptyList(),

    @Indexed
    val timestamp: Instant = Instant.now()
)

// Вспомогательный класс для вложенного документа
data class ChatAttachment(
    val contentType: String, // например, "image/jpeg"
    val data: Binary          // Сами байты картинки
)